optuna/samplers/nsgaii/_sampler.py

```diff
@@ -485,6 +485,9 @@ def _constrained_dominates(
     assert isinstance(constraints0, (list, tuple))
     assert isinstance(constraints1, (list, tuple))
 
+    if np.any(np.isnan(constraints0)) or np.any(np.isnan(constraints1)):
+        raise ValueError("NaN is not acceptable as constraint value.")
```

Comment:
[question] For example, only constraints1 has nan, can we assume that constraints0 is feasible? like follows

```
    if constraints0 is not None and constraints1 is None:
        # Trial x has constraint values, but y doesn't.
        return True
```

Reply:
I think having `nan` in constraints means there is a bug in the user code, and we should notify the user as early as possible. That's why I didn't consider `nan` constraint as an infeasible trial but reported error.